scripts/generate_image_manifest.py

```
@@ -0,0 +1,119 @@
+    for component, class_path in sorted(IMAGE_SOURCES.items()):
+        module_path, class_name = class_path.rsplit(".", 1)
+        module = importlib.import_module(name=module_path)
+        cls = getattr(module, class_name)
+        images = sorted(set(_extract_images(cls=cls)))
+        if images:
+            manifest[component] = images
+    return manifest
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)  # noqa: FCN001
```

Ruff (0.15.21)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)